### PR TITLE
oauth: Ensure we sync newly created external services

### PR DESCRIPTION
**Pull request by ryanslade**
We ensure that we call out to repo-updater to sync newly added external services that are
created during our OAuth flow.

The `syncExternalService` helper was moved from `graphqlbackend` to the `backend` package
so that it could be called from our OAuth handler too.

# Test plan

Tests pass and tested the flow on a local instance

### CODE REVIEW

File: enterprise/cmd/frontend/internal/auth/oauth/session.go [hunk 1]
```diff
@@ -71,13 +74,18 @@ func SessionIssuer(db database.DB, s SessionIssuerHelper, sessionKey string) htt
...
 		if state.Op == LoginStateOpCreateCodeHostConnection {
-			safeErrMsg, err := s.CreateCodeHostConnection(ctx, token, state.ProviderID)
+			svc, safeErrMsg, err := s.CreateCodeHostConnection(ctx, token, state.ProviderID)
 			if err != nil {
 				log15.Error("OAuth failed: error upserting code host connection from OAuth token.", "error", err, "userErr", safeErrMsg)
 				http.Error(w, safeErrMsg, http.StatusInternalServerError)
 				return
 			}
 
+			if err := backend.SyncExternalService(ctx, svc, 5*time.Second, repoupdater.DefaultClient); err != nil {
```
Comment (ryanslade) [inline enterprise/cmd/frontend/internal/auth/oauth/session.go:84]: I just went with the default timeout here.

Comment (indradhanush) [reply enterprise/cmd/frontend/internal/auth/oauth/session.go:84]: I couldn't spot where we have the 5s as the default timeout, but maybe we can add this as a comment?

**Review by indradhanush — APPROVED** [review]

File: enterprise/cmd/frontend/internal/auth/oauth/session.go [hunk 2]
```diff
@@ -71,13 +74,18 @@ func SessionIssuer(db database.DB, s SessionIssuerHelper, sessionKey string) htt
...
+				log15.Error("OAuth failed: error syncing external service", "error", err)
+				http.Error(w, "error syncing code host", http.StatusInternalServerError)
+			}
```
Comment (unknwon) [inline enterprise/cmd/frontend/internal/auth/oauth/session.go:87]: Are we missing a return here?

Comment (ryanslade) [reply enterprise/cmd/frontend/internal/auth/oauth/session.go:87]: Yes, good spot!

**Pull request merged**